smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelValidator.java

```diff
@@ -214,6 +245,39 @@ public Validator createValidator() {
         };
     }
 
+    /**
+     * Applies each of the given decorators to each of the validation events and returns the resulting list of
+     * validation events.
+     *
+     * @param decorators The decorators to apply
+     * @param events The events to decorate
+     * @return The decorated events
+     */
+    public static List<ValidationEvent> decorateEvents(
+        List<ValidationEventDecorator> decorators,
+        List<ValidationEvent> events
+    ) {
+        if (!decorators.isEmpty()) {
+            try {
+                List<ValidationEvent> decorated = new ArrayList<>(events.size());
+                for (ValidationEvent event : events) {
+                    ValidationEvent decoratedEvent = event;
+                    for (ValidationEventDecorator decorator : decorators) {
+                        if (decorator.canDecorate(event)) {
+                            decoratedEvent = decorator.decorate(event);
+                        }
+                    }
+                    decorated.add(decoratedEvent);
+                }
+                return decorated;
+            } catch (Throwable e) {
```

Comment:
Will this be a common issue? Why don't we expose these exceptions to the end user when they occur?

Reply:
I don't know if this will be common and we could expose the exceptions, however I'm not sure if breaking the build in potentially hard to solve way (e.g., the library is already part of the build dependency) is going to be a good user experience.